modules/internal/DefaultRectangular.chpl

```diff
@@ -2143,10 +2171,10 @@ module DefaultRectangular {
 
     if enableParallelGetsInAssignment || enableParallelPutsInAssignment {
       if isSizeAboveThreshold && !isFullyLocal {
-        if enableParallelPutsInAssignment && Blocid == here.id {
+        if enableParallelPutsInAssignment && _isLocSulocSameAsHere(Blocid, Bsublocid) {
           doParallelAssign = true;
         }
-        else if enableParallelGetsInAssignment && Blocid != here.id {
+        else if enableParallelGetsInAssignment && !_isLocSulocSameAsHere(Blocid, Bsublocid) {
           doParallelAssign = true;
```

Comment:
these have typos, I am surprised this compiles

Reply:
Ah, because `enableParallel*InAssignment` is a param. Consider adding this to your test matrix

Reply:
Suspect it's because `enableParallelPutsInAssignment` etc. are off